sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -115,6 +116,17 @@ private[sql] case class PreWriteCheck(catalog: Catalog) extends (LogicalPlan =>
           // OK
         }
 
+        // Get all input data source relations of the query.
+        val srcRelations = query.collect {
```

Comment:
Actually we can do that in another PR, as will check if the the overwritten table is the same as the source table for the `HadoopFsRelation` based data source.

This actually will break the json unit test if we don't change the code in this PR.